src/core/ConfigManager.cpp

```diff
@@ -347,18 +346,7 @@ const QString & ConfigManager::value(const QString & cls,
 			}
 		}
 	}
-	static QString empty;
-	return empty;
-}
-
-
-
-const QString & ConfigManager::value(const QString & cls,
-				      const QString & attribute,
-				      const QString & defaultVal) const
-{
-	const QString & val = value(cls, attribute);
-	return val.isEmpty() ? defaultVal : val;
```

Comment:
The new behaviour is subtly different: if the value is present but empty, the old code returns the default, but the new code returns the empty value. I'm not sure whether this is actually a problem, but it's something to be aware of.

Reply:
> The new behaviour is subtly different: if the value is present but empty, the old code returns the default, but the new code returns the empty value. I'm not sure whether this is actually a problem, but it's something to be aware of.

I'm not sure either, but I will err on the side of caution and try to match with the original behavior. Changing the behavior of how the code works isn't intended in this PR anyway.

Reply:
So usually I would stay on the safe side with this, but I've given it more thought. If the value was found but empty, wouldn't it be more reasonable to return that empty value? An attribute set to an empty value in the configuration is still a valid value. Returning ``defaultVal`` would make sense if that attribute was not found, which is what is done here.